src/main/groovy/jenkins/automation/utils/ScmUtils.groovy

```diff
@@ -24,8 +32,9 @@ class ScmUtils {
                         branch("*/tags/$version")
                     }
                     relativeTargetDir(repo.sub_directory)
+                    shallowClone(shallow)
```

Comment:
repo.shallow will be null if the shallow key isn't provided, and shallowClone needs a boolean argument.